RevenueCatUI/Views/GradientView.swift

```diff
@@ -0,0 +1,84 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  GradientView.swift
+//
+//  Created by Mark Villacampa on 2024-11-24.
+
+import SwiftUI
+
+#if PAYWALL_COMPONENTS
+
+struct GradientView: View {
+    enum GradientStyle {
+        case linear(Int)
+        case radial
+    }
+
+    @Environment(\.colorScheme)
+    private var colorScheme
+
+    let lightGradient: Gradient
+    let darkGradient: Gradient?
+    let gradientStyle: GradientStyle
+
+    private var gradient: Gradient {
+        switch colorScheme {
+        case .light:
+            lightGradient
+        case .dark:
+            darkGradient ?? lightGradient
+        @unknown default:
+            lightGradient
+        }
+    }
+
+    var body: some View {
+        switch gradientStyle {
+        case .linear(let degrees):
+            LinearGradient(
+                gradient: gradient,
+                startPoint: UnitPoint(angle: Angle(degrees: Double(degrees))),
+                endPoint: UnitPoint(angle: Angle(degrees: Double(degrees+180)))
+            )
+        case .radial:
+            RadialGradient(
+                gradient: gradient,
+                center: .center,
+                startRadius: 0,
+                endRadius: 100
+            )
+        }
+    }
+}
+
+private extension UnitPoint {
+
+    init(angle: Angle) {
+        // Convert the angle to radians
+        let radians = angle.radians
+
+        // Calculate the normalized x and y positions
+        let xPosition = cos(radians)
+        let yPosition = sin(radians)
+
+        // Determine the scaling factor to move the point to the edge of the enclosing square
+        let scaleFactor = max(abs(xPosition), abs(yPosition))
```

Comment:
Thanks a lot for the inspiration here! This was very useful for Android too. 🙏 

Quick question: does this work for non-square shapes in SwiftUI, like rectangles? In Compose, I find that the angle isn't quite correct when using a rectangle, as it indeed seems to "pretend" it's a square and draws the angle based on that. Not sure how strict we are with the desired result. There's a bunch of discussion and suggested Compose solutions on [this StackOverflow question](https://stackoverflow.com/q/68218714).

Here's an example of what I mean. Both are set to an angle of 45 degrees, but only the square is actually 45 degrees.
<img src="https://github.com/user-attachments/assets/de8d804e-b6ed-49cd-95fb-86a13b9666db" width="400"  />

Reply:
Will handle in a new PR 👍